runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -314,13 +332,15 @@ public void append(K key, W window, Iterator<V> values) {
             while (values.hasNext()) {
               bagState.add(values.next());
             }
+
+            return generateAndRegisterCacheKey();
```

Comment:
In some scenarios runner wants sdk worker to not cache the value. It indicates it by not passing any cache token. How are we addressing that scenario?

Reply:
I cannot think of any scenarios for user state where we wouldn't want to cache. It is important though that the cache tokens can be cleared such that they are not sent to the SDK anymore, e.g. when restoring from a checkpoint.

Reply:
Note that this is possible now because the cache token is passed via an `Optional`.